File: app/src/main/java/com/example/lostandfound/ui/profile/ProfileViewModel.java
package com.example.lostandfound.ui.profile;

import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ProfileViewModel extends ViewModel {

    /*
    Link the textview to the count that is stored in the viewmodel.

    Whenever that count gets updated, an observer is set up to also change
    the displayed value (the binding object).

    The observer is set in the Activity file.
     */
    private final MutableLiveData<Integer> count = new MutableLiveData<>(0);   // set the initial value here

    public LiveData<Integer> getCount(){
        return count;
    }

    public void updateCount(int newCount){
        count.setValue(newCount);
    }

}

File: app/src/main/java/com/example/lostandfound/ui/Profile/ProfileViewModel.kt
package com.example.lostandfound.ui.Profile

import android.net.Uri
import androidx.compose.runtime.MutableState
import androidx.compose.runtime.mutableStateOf
import androidx.lifecycle.ViewModel

class ProfileViewModel : ViewModel(){
    val imageUri: MutableState<Uri?> = mutableStateOf(null)
    val email: MutableState<String> = mutableStateOf("E")
    val firstName: MutableState<String> = mutableStateOf("se")
    val lastName: MutableState<String> = mutableStateOf("e")

    fun onImageChanged(uri: Uri?){
        imageUri.value = uri
    }

    fun onEmailChanged(s: String){
        email.value = s
    }

    fun onFirstNameChanged(s: String){
        firstName.value = s
    }

    fun onLastNameChanged(s: String){
        lastName.value = s
    }
}